### ISSUE TITLE
Automatically create a deployment for Pull request 

### ISSUE THREAD

**Issue by jainkeval** [probot/ideas#67]
This app will be for frontend projects. The bot will deploy site to heroku (may be) for one's pull request. So that the mentors will be able to review the changes easily.

@bkeepers @gr2m any suggestions?

I would like to work on this.

**Comment by itaditya** [comment]
if you are hosting only frontend then there are easier options available than heroku. Check surge.sh

**Comment by maracuja-juice** [comment]
@itaditya That's correct! Some more options for the Frontend are Github Pages and Netlify.

**Comment by kenorb** [comment]
Why not to use [Travis CI](https://docs.travis-ci.com/user/deployment/), it can deploy to Heroku as well.

**Comment by patcon** [comment]
I like the idea, but unless I'm misunderstanding, I do this with Heroku Review Apps already:
https://devcenter.heroku.com/articles/github-integration-review-apps

Review apps importantly only work on PRs from branches in the upstream repo, not forks. But with protected branches, maintainers can liberally give push access without worrying about `master`, and so the workflow works great.

Plus `probot/settings` is about the allow restricted branches to be configured from file, so even simpler to share this practice: https://github.com/probot/settings/pull/76
